soccer/src/vision/VisionFilter.cpp

```diff
@@ -91,27 +115,27 @@ void VisionFilter::updateLoop() {
 
         {
             // Do update with whatever is in frame buffer
-            std::lock_guard<std::mutex> lock1(frameLock);
-            {
-                std::lock_guard<std::mutex> lock2(worldLock);
-
-                if (!frameBuffer.empty()) {
-                    world.updateWithCameraFrame(RJ::now(), frameBuffer);
-                    frameBuffer.clear();
-                } else {
-                    world.updateWithoutCameraFrame(RJ::now());
-                }
+            GetFrames();
+            std::lock_guard<std::mutex> lock2(worldLock);
+
+            if (!frameBuffer.empty()) {
+                world.updateWithCameraFrame(RJ::now(), frameBuffer);
+                frameBuffer.clear();
+            } else {
+                world.updateWithoutCameraFrame(RJ::now());
             }
         }
 
         // Wait for the correct loop timings
         RJ::Seconds diff = RJ::now() - start;
-        RJ::Seconds sleepLeft = RJ::Seconds(*VisionFilterConfig::vision_loop_dt) - diff;
+        RJ::Seconds sleepLeft =
+            RJ::Seconds(*VisionFilterConfig::vision_loop_dt) - diff;
 
         if (diff > RJ::Seconds(0)) {
             std::this_thread::sleep_for(sleepLeft);
```

Comment:
`std::this_thread::sleep_until` is what you want here.

Reply:
I'd rather keep this PR related `vision_receiver` as much as possible and avoid changing `VisionFilter` related functionality even if its bad, and leave that for a different PR just to keep this one focused and small.

Reply:
Oh sure, that's fine. Just add a TODO.